src/utils.rs

```diff
@@ -189,22 +196,86 @@ pub fn collect_prob_dist<E: Event>(
             }
         }
 
-        let variants = try!(utils::collect_variants(&mut record, false, false, None, false));
-        let tag = event.tag_name("PROB");
-        let event_probs = try!(record.info(tag.as_bytes()).float());
-        if let Some(event_probs) = event_probs {
-            // tag present
-            for (variant, event_prob) in variants.into_iter().zip(event_probs.into_iter()) {
-                if let Some(variant) = variant {
-                    if !variant.is_type(vartype) || event_prob.is_nan() {
-                        continue;
+        let variants = (utils::collect_variants(&mut record, false, false, None, false))?;
+        let mut events_prob_sum = LogProb::ln_zero();
+        for tag in &tags {
+            if let Some(event_probs) = (record.info(tag.as_bytes()).float())? {
```

Comment:
I think the parentheses are superfluous. But this is so minor that we do not need to handle this now.